src/gpu.rs

```diff
@@ -169,4 +177,20 @@ mod tests {
     fn smoke_detector() {
         StandardGpuResources::new().unwrap();
     }
+
+    // The test marked as ignored because it takes a significant amount of time.
+    #[ignore]
+    #[test]
+    fn resources_leak() {
+        use crate::{index_factory, MetricType};
+
+        // Try to allocate gpu resources multiple times in a loop.
+        // If resources are not properly deallocated this will cause out-of-memory error.
+        for _ in 0..50 {
```

Comment:
This test can be considered just as reproducer of the issue. It should not be run every time, or maybe it should be removed at all because it consumes a lot of resources (all GPU memory) just to ensure that destructor is called.

Reply:
I'll go ahead and merge as-is, because there is not that much mental overhead in having an ignored test.